src/app/component/general/eco-news/news-list/news-list.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, OnInit, Input, OnDestroy } from '@angular/core';
+import { EcoNewsService } from 'src/app/service/eco-news/eco-news.service';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-news-list',
+  templateUrl: './news-list.component.html',
+  styleUrls: ['./news-list.component.css']
+})
+export class NewsListComponent implements OnInit, OnDestroy {
+  private view: boolean;
+  private iterator: number;
+  private toggler = false;
+
+  @Input() gridOutput: Array<string>;
+
+  private ecoNewsSubscription: Subscription;
+  private allEcoNews = [];
+  private elements = [];
+  public remaining = 0;
+
+  constructor(private ecoNewsService: EcoNewsService) { }
+
+  ngOnInit() {
+    this.fetchAllEcoNews();
+  }
+  private fetchAllEcoNews(): void {
+    this.ecoNewsSubscription = this.ecoNewsService
+      .getAllEcoNews()
+      .subscribe(data => {
```

Comment:
Please create a separate(create method and replace logic from subscribe to this method) callback for the subscribe

Reply:
Fixed